internal/backend/remote/backend.go

```diff
@@ -679,20 +675,6 @@ func (b *Remote) StateMgr(ctx context.Context, name string) (statemgr.Full, erro
 		}
 	}
 
-	// This is a fallback error check. Most code paths should use other
-	// mechanisms to check the version, then set the ignoreVersionConflict
-	// field to true. This check is only in place to ensure that we don't
-	// accidentally upgrade state with a new code path, and the version check
-	// logic is coarser and simpler.
-	if !b.ignoreVersionConflict {
-		wsv := workspace.TerraformVersion
-		// Explicitly ignore the pseudo-version "latest" here, as it will cause
-		// plan and apply to always fail.
-		if wsv != tfversion.String() && wsv != "latest" {
-			return nil, fmt.Errorf("Remote workspace OpenTofu version %q does not match local OpenTofu version %q", workspace.TerraformVersion, tfversion.String())
-		}
-	}
```

Comment:
I still think we want this check, we don't want someone mixing versions up. 

in this case i think we should still throw an error if someone has a workspace version of 1.6 and a tfversion of 1.9 (for example).

Reply:
@kislerdm and I are on a call right now puzzling out where "latest" is supposedly set.

Reply:
Our hypothesis is that the remote workspace is supposed to default to "latest" when terraform-version is not set.  JFrog's Artifactory probably does not include this implementation Quirk.

Reply:
There are other overrides for version = "latest" in other parts of the workspace code that @kislerdm identified

Reply:
IMO terraform has weird hacks around alphas, which is what is biting us at the moment.

Reply:
@Yantrio updated. Please have a look.